lib/virt_feature_test_base.pm

```diff
@@ -157,7 +158,21 @@ sub post_fail_hook {
     my ($self) = shift;
 
     $self->{"stop_run"} = time();
-    assert_script_run("history -w /root/commands_history");
+    unless (defined(script_run("history -w /root/commands_history", die_on_timeout => 0))) {
+        # Switch to sol console to check serial console output
+        # It is useful in the case of host crash and reboot
+        record_info("WARN", "ssh connection is broken and switch to SOL console", result => 'softfail');
+        select_console 'sol', await_console => 0;
+        save_screenshot;
+        wait_still_screen 5;
+        check_screen("linux-login", 120);
+        reset_consoles;
+        record_info("WARN", "switch back to ssh console to collect logs", result => 'softfail');
+        select_console('root-ssh');
```

Comment:
I think it does not make sense by calling select_console('root-ssh') in sol console in this case.

Reply:
`select_console('root-ssh')` is to change to ssh console from sol console.

Reply:
But as you said, this block is executed if ssh connection is lost, so why bother ```select_console('root-ssh')``` ?

Reply:
No, ssh console is not dead. when we detected the ssh session dropped firstly by 'defined(script_run())',  we change to use sol console, then reconnect ssh console by 'select_console('root-ssh')'. The ssh connection comes back online again. I had several successfuly cases, fg. the verification run \#1 and \#3.